docs/book/v4/coding-style-guide.md

```diff
@@ -201,49 +204,108 @@ Any new types and keywords added to future PHP versions MUST be in lower case.
 Short form of type keywords MUST be used i.e. `bool` instead of `boolean`,
 `int` instead of `integer` etc.
 
-### 2.6 Variables
+Compound types includes intersection, union, and mixed intersection and union type declarations. PHP requires
+that all compound types be structured as an ORed (unioned) series of ANDs (intersections), and that each set of
+intersections be encased with parentheses.
 
-> ### Additional Laminas rules
->
-> Variable names MUST be declared in camelCase.
+The union symbol `|` and intersection symbol `&` MUST NOT have a leading or trailing space.  The parentheses MUST NOT
+have a leading or trailing space.
 
-### 2.7 Arrays
+If it is necessary to split a compound type into multiple lines:
 
-> ### Additional Laminas rules
->
-> The short array syntax MUST be used to define arrays.
->
-> All values in multiline arrays must be indented with 4 spaces.
->
-> All array values must be followed by a comma, including the last value.
->
-> There MUST NOT be whitespace around the opening bracket or before the closing
-> bracket when referencing an array.
->
-> All double arrow symbols MUST be aligned to one space after the longest array
-> key.
+- If the type contains only intersections or only unions, then each line MUST have a single type.
+- If the type contains both intersections and unions, then each line MUST have a single union segment. All intersections in a segment MUST be on the same line.
+- The symbol on which the compound type is split MUST be at the start of each line.
+
+The following are correct ways to format compound types:
 
 ```php
-$array2 = [
-    'one'    => function () {
-        $foo    = [1, 2, 3];
-        $barBar = [
-            1,
-            2,
-            3,
-        ];
-    },
-    'longer' => 2,
-    3        => 'three',
-];
+function foo(int|string $a): User|Product
+{
+    // ...
+}
+
+function somethingWithReflection(
+    \ReflectionObject
+    |\ReflectionClass
+    |\ReflectionMethod
+    |\ReflectionParameter
+    |\ReflectionProperty $reflect
+): object|null {
+        // ...
+}
+
+function complex(array|(ArrayAccess&Traversable) $input): ArrayAccess&Traversable
+{
+    // ...
+}
+
+function veryComplex(
+    array
+    |(ArrayAccess&Traversable)
+    |(Traversable&Countable) $input): ArrayAccess&Traversable
```

Comment:
This appears to contradict the rest of the standard. I would think that the closing parentheses would need to be on a new line along with the return type and opening brace?

```php
//...
$input
): ArrayAccess&Traversable {
```